blockchain/blockchain.go

```diff
@@ -696,7 +696,7 @@ func (bc *blockchain) commitBlock(blk *Block) error {
 	if bc.sf != nil {
 		ExecuteContracts(blk, bc)
 		if err := bc.sf.CommitStateChanges(blk.Height(), blk.Transfers, blk.Votes, blk.Executions); err != nil {
-			return err
+			logger.Fatal().Err(err).Msgf("Failed to commit state changes on height %d", blk.Height())
```

Comment:
we should also Fatal() if bc.dao.putBlock() != nil above?

Reply:
IMHO, it should be fine for putBlock.If it fails, nothing else will be persisted. The blockchain will stuck at a block height, so that we could easily see.

Reply:
But putReceipts may need, which is yet another separate batch after block is persisted